src/status_im/contexts/profile/settings/screens/password/change_password/effects.cljs

```diff
@@ -0,0 +1,18 @@
+(ns status-im.contexts.profile.settings.screens.password.change-password.effects
+  (:require [native-module.core :as native-module]
+            [promesa.core :as promesa]
+            [status-im.common.keychain.events :as keychain]
+            [utils.re-frame :as rf]
+            [utils.security.core :as security]))
+
+(rf/reg-fx
+ :effects.change-password/change-password
+ (fn [{:keys [key-uid old-password new-password on-success on-fail]}]
+   (let [hash-masked-password (fn [pw]
+                                (-> pw security/safe-unmask-data native-module/sha3))
+         old-password-hashed  (hash-masked-password old-password)
+         new-password-hashed  (hash-masked-password new-password)]
+     (-> (native-module/reset-password key-uid old-password-hashed new-password-hashed)
+         (promesa/then #(keychain/save-user-password! key-uid new-password-hashed))
```

Comment:
`promesa` as an alias looks so verbose. I see we changed the alias here 53874c8343ef173a43a089916aaf4008f0c0fdf4 🤷🏼 Okay... I hope we can rename it soon to just `p` as it's the alias almost everybody uses with promesa.